ext/src/ruby_api/engine.rs

```diff
@@ -1,15 +1,35 @@
 use super::{config::Config, root};
 use crate::error;
+use async_timer::Interval;
+use lazy_static::lazy_static;
 use magnus::{function, method, scan_args, Error, Module, Object, RString, Value};
+use std::cell::RefCell;
+use tokio::{runtime, task::JoinHandle, time};
 use wasmtime::Engine as EngineImpl;
 
+lazy_static! {
+    static ref TOKIO_RT: runtime::Runtime = runtime::Builder::new_multi_thread()
+        .thread_name("wasmtime-engine-timers")
+        .worker_threads(1)
+        .enable_time()
+        .enable_io()
+        .build()
+        .unwrap();
+}
+
 /// @yard
 /// Represents a Wasmtime execution engine.
 /// @see https://docs.rs/wasmtime/latest/wasmtime/struct.Engine.html Wasmtime's Rust doc
-#[derive(Clone)]
-#[magnus::wrap(class = "Wasmtime::Engine")]
+#[magnus::wrap(class = "Wasmtime::Engine", free_immediately)]
 pub struct Engine {
     inner: EngineImpl,
+    timer_task: RefCell<Option<JoinHandle<()>>>,
+}
+
+impl Drop for Engine {
+    fn drop(&mut self) {
+        self.stop_epoch_timer()
+    }
```

Comment:
There is no test for that -- couldn't find a sane way of testing: can't read the engine after it's gone.

I did test this manually by adding print statements when ticking, and confirmed that on GC the ticking stopped.